policy/policy_any.c

```diff
@@ -278,229 +279,248 @@ get_max_priority(const GSList *existing_requests)
-        {
-          ILOG_DEBUG("service module found for network type '%s'",
-                     (char *)network_types_old->data);
-          network_types = g_slist_prepend(network_types,
-                                          network_types_old->data);
-        }
-        else
-        {
-          ILOG_DEBUG("network type '%s' IAPs not found in gconf",
-                     (char *)network_types_old->data);
-          g_free(network_types_old->data);
-        }
-
-        network_types_old = g_slist_delete_link(network_types_old,
-                                                network_types_old);
-      }
+      scan_types_list_old = g_slist_delete_link(scan_types_list_old,
+					      scan_types_list_old);
     }
   }
 
-skip:
   g_object_unref(gconf);
+  return scan_types_list;
+}
 
-  data->network_types = network_types;
-  data->max_priority = get_max_priority(existing_requests);
+static void
+policy_any_new_request(struct icd_policy_request *new_request,
+		       const GSList *existing_requests,
+		       icd_policy_request_new_cb_fn policy_done_cb,
+		       gpointer policy_token, gpointer *private)
+{
+  struct policy_any_data *data = (struct policy_any_data *)*private;
+  gboolean scan_started;
+
+  if (strcmp(new_request->network_id, OSSO_IAP_ANY))
+  {
+    policy_done_cb(ICD_POLICY_ACCEPTED, new_request, policy_token);
+    return;
+  }
+
+  if (new_request->attrs & ICD_POLICY_ATTRIBUTE_HAS_CONNECTIONS)
+  {
+    ILOG_INFO("any connection request %p already has connections", new_request);
+    policy_done_cb(ICD_POLICY_ACCEPTED, new_request, policy_token);
+    return;
+  }
+
+  if (data->ongoing_scans)
+  {
+    GSList *l;
+
+    for (l = data->ongoing_scans; l; l = l->next)
+    {
+      ILOG_ERR("any connection waiting for OSSO_IAP_ANY request %p scan_data %p reply",
+               data->request->request_token, l->data);
+    }
+
+    policy_done_cb(ICD_POLICY_REJECTED, new_request, policy_token);
+
+    return;
+  }
+
+  data->request = new_request;
+  data->scan_types_list = policy_any_get_types(data);
+  data->min_prio = policy_any_get_prio(existing_requests);
 
   ILOG_DEBUG("any connection request %p scanning for networks with prio > %d",
-             data->request->request_token, data->max_priority);
+             data->request->request_token, data->min_prio);
 
   scan_started = FALSE;
 
-  while(data->network_types)
+  while(data->scan_types_list)
   {
-    gchar *type = (gchar *)data->network_types->data;
-    data->network_types = g_slist_delete_link(data->network_types,
-                                              data->network_types);
+    gchar *type = (gchar *)data->scan_types_list->data;
+    data->scan_types_list = g_slist_delete_link(data->scan_types_list,
+                                              data->scan_types_list);
 
     if (type && *type)
     {
-      struct scan_data *scan_data;
-
+      struct policy_scan_data *scan_data = policy_any_scan_data_new(
```

Comment:
not really a good idea to mix renaming to code changes, it becomes very hard to review